### ISSUE TITLE
filter_stack add name preview causes a crash with unhandled exception on bad regex

### ISSUE THREAD

**Issue by bothers** [ranger/ranger#2848]
#### Runtime Environment

- Operating system and version:  Linux 6.3.2-arch1-1
- Terminal emulator and version:  kitty 0.28.1
- Python version: 3.11.3
- Ranger version/commit: fe7c3b2
- Locale: en_US.UTF-8

#### Current Behavior
Typing bad regex after `:filter_stack add name` causes Ranger to crash with an unhandled re.error exception.

#### Expected Behavior
Not crashing, but maybe some indication that compilation failed would be nice.

#### Possible Solutions
I'm unfamiliar with Python, but adding a try/catch block around [these lines](https://github.com/ranger/ranger/blob/master/ranger/config/commands.py#L1829-L1830) stops Ranger from crashing.  Whether or not silently failing and not updating the preview is the best way to handle this situation is another issue.

#### Steps to reproduce
1. Type `.n*` or `:filter_stack add name *`

#### Traceback

```
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/ranger/core/main.py", line 214, in main
    fm.loop()
  File "/usr/lib/python3.11/site-packages/ranger/core/fm.py", line 423, in loop
    ui.handle_input()
  File "/usr/lib/python3.11/site-packages/ranger/gui/ui.py", line 276, in handle_input
    self.handle_key(key)
  File "/usr/lib/python3.11/site-packages/ranger/gui/ui.py", line 206, in handle_key
    elif not DisplayableContainer.press(self, key):
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/ranger/gui/displayable.py", line 283, in press
    focused_obj.press(key)
  File "/usr/lib/python3.11/site-packages/ranger/gui/widgets/console.py", line 193, in press
    self.type_key(key)
  File "/usr/lib/python3.11/site-packages/ranger/gui/widgets/console.py", line 222, in type_key
    self.on_line_change()
  File "/usr/lib/python3.11/site-packages/ranger/gui/widgets/console.py", line 588, in on_line_change
    if cmd and cmd.quick():
               ^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/ranger/config/commands.py", line 1829, in quick
    self.fm.thisdir.temporary_filter = re.compile(self.rest(3))
                                       ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/re/__init__.py", line 227, in compile
    return _compile(pattern, flags)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/re/__init__.py", line 294, in _compile
    p = _compiler.compile(pattern, flags)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/re/_compiler.py", line 743, in compile
    p = _parser.parse(p, flags)
        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/re/_parser.py", line 980, in parse
    p = _parse_sub(source, state, flags & SRE_FLAG_VERBOSE, 0)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/re/_parser.py", line 455, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/re/_parser.py", line 682, in _parse
    raise source.error("nothing to repeat",
re.error: nothing to repeat at position 0
```

**Comment by toonn** [comment]
You're right, I noticed this too but I must've forgotten to open an issue. Simply dropping the exception is an OK fix but I'm also considering dropping characters from the RE until there's no exception, that way behavior would be consistent whether you're typing a pattern or pasting it all at once.

**Comment by eMPee584** [comment]
> I'm also considering dropping characters from the RE until there's no exception

Why fudge with incomplete / invalid user input, just don't crash... 😅
Crash can also be triggered entering `add name \` => `re.error: bad escape (end of pattern) at position 0` ..

Following seems to work fine and the actual regex problem is given to the user when confirming with `[ENTER]`.

```
        if self.rest(1).startswith("add name "):
            try:
                self.fm.thisdir.temporary_filter = re.compile(self.rest(3))
            except re.error:
                pass
            else:
                self.fm.thisdir.refilter()

```

**Comment by vifon** [comment]
Fixed in 38bb890.
